### PR TITLE
feat(MRS): support external data sources

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

1. the acc test used a resource: huaweicloud_mapreduce_connection which is not ready currently.
2. The test result is based on some manually create resources: rds_mysql, database, mrs_connection

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccMrsMapReduceCluster_externalDataSources$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/mrs -v

=== RUN   TestAccMrsMapReduceCluster_externalDataSources
=== PAUSE TestAccMrsMapReduceCluster_externalDataSources
=== CONT  TestAccMrsMapReduceCluster_externalDataSources
--- PASS: TestAccMrsMapReduceCluster_externalDataSources (1086.42s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/mrs	1086.460s
```

### CODE REVIEW

File: docs/resources/mapreduce_cluster.md [hunk 1]
```diff
@@ -512,6 +518,30 @@ The `configs` block supports:
 * `config_file_name` - (Required, String, ForceNew) Specifies the configuration file name of component installed.
   Changing this will create a new MapReduce cluster resource.
 
+<a name="ExternalDatasources"></a>
+The `external_datasources` block supports:
+
+* `component_name` - (Required, String, ForceNew) Specifies the Component name. The valid values are `Hive` and `Ranger`.
```
Comment (ShiChangkuo) [inline docs/resources/mapreduce_cluster.md]:
```suggestion
* `component_name` - (Required, String, ForceNew) Specifies the component name. The valid values are `Hive` and `Ranger`.
```

also missing `Changing this ...`

File: huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_cluster_test.go [hunk 2]
```diff
@@ -891,6 +930,94 @@ master_nodes {
...
+resource "huaweicloud_mapreduce_cluster" "test" {
+  availability_zone  = data.huaweicloud_availability_zones.test.names[0]
+  name               = "%[2]s"
+  type               = "ANALYSIS"
+  version            = "MRS 1.9.2"
+  safe_mode          = false
+  manager_admin_pass = "%[3]s"
+  node_admin_pass    = "%[3]s"
+  vpc_id             = "60f6bd6b-3b21-4f07-b5e0-f2f876410902" //huaweicloud_vpc.test.id
+  subnet_id          = "828907cc-40c9-42fe-8206-ecc1bdd30060" //huaweicloud_vpc_subnet.test.id
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_cluster_test.go]:
```suggestion
  vpc_id             = huaweicloud_vpc.test.id
  subnet_id          = huaweicloud_vpc_subnet.test.id
```

File: huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_cluster_test.go [hunk 3]
```diff
@@ -891,6 +930,94 @@ master_nodes {
...
+    data_volume_type  = "SAS"
+    data_volume_size  = 600
+    data_volume_count = 1
+  }
+
+  external_datasources {
+    component_name = "Hive"
+    role_type      = "hive_metastore"
+    source_type    = "RDS_MYSQL"
+    connector_id   = huaweicloud_mapreduce_connection.test.id
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_cluster_test.go]: replace with `HW_MAPREDUCE_EXTERNAL_DATASOURCES` ?

File: docs/resources/mapreduce_cluster.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/mapreduce_cluster.md]: done

File: huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_cluster_test.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_cluster_test.go]: done

File: huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_cluster_test.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_cluster_test.go]: huaweicloud_mapreduce_connection is declared above, which is not ready currently. 
connection and cluster must be in the same vpc , so cannot create it manually.

File: huaweicloud/services/mrs/resource_huaweicloud_mapreduce_cluster.go [hunk 4]
```diff
@@ -382,6 +388,33 @@ func componentConfigsSchemaResource() *schema.Resource {
...
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"source_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"connector_id": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/mrs/resource_huaweicloud_mapreduce_cluster.go]: use `connection_id` will be better as the resource name is `xxx_data_connection`

Comment (hmbig2) [reply huaweicloud/services/mrs/resource_huaweicloud_mapreduce_cluster.go]: ok, change to data_connection_id, and optinal

**Pull request merged**